handshake_ix.go

```diff
@@ -350,22 +340,6 @@ func ixHandshakeStage2(f *Interface, addr *udp.Addr, via *ViaSender, hostinfo *H
 	}
 
 	ci := hostinfo.ConnectionState
-	if ci.ready {
```

Comment:
if we remove this ci.ready check, I think we need a different double check after the `hh.Lock()` to see if a different thread already completed the handshake for this host? How do we protected about late handshake packets coming in?

Reply:
I expect in this version it would fail when trying to read a noise message. The next PR in the series would put these calls to `ixHandshake*` within the pending hostmap lock to avoid these races.